### PR TITLE
[FLINK-25685][rest] RestClusterClient supports uploading local file path with scheme

### PR DESCRIPTION
**Pull request by Sxnan**
## What is the purpose of the change

Fix the problem that RestClusterClient gets stuck on submitting job with local user artifact.

## Brief change log

- Change how Path is constructed to support local file path with scheme.

## Verifying this change

This change added tests and can be verified as follows:

- Added test that validates that local file path with scheme is supported.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector: no
## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable

### CODE REVIEW

File: flink-clients/src/main/java/org/apache/flink/client/program/rest/RestClusterClient.java [hunk 1]
```diff
@@ -357,7 +357,8 @@ public void close() {
                                                         artifactFilePath.getName()));
                                         filesToUpload.add(
                                                 new FileUpload(
-                                                        Paths.get(artifacts.getValue().filePath),
+                                                        Paths.get(
+                                                                artifactFilePath.toUri().getPath()),
```
Comment (zentol) [inline flink-clients/src/main/java/org/apache/flink/client/program/rest/RestClusterClient.java:361]:
```suggestion
                                                                artifactFilePath.toUri()),
```
This seems redundant given that we do `Paths.get`?

Comment (Sxnan) [reply flink-clients/src/main/java/org/apache/flink/client/program/rest/RestClusterClient.java:361]: If the `URI` of the artifactFilePath doesn't have a scheme, the `Paths.get` throws an `IllegalArgumentException` exception. In order to support both with and without schema, we have to use the `Paths.get` that takes a file path string.

By looking at the code, I notice that `artifactFilePath.toUri().getPath()` is the same as `artifactFilePath.getPath()`. I think we can use `artifactFilePath.getPath()`. What do you think?

Comment (zentol) [reply flink-clients/src/main/java/org/apache/flink/client/program/rest/RestClusterClient.java:361]: Ah, URI#getPath returns a string.

> we can use artifactFilePath.getPath()

sounds good 👍

Comment (Sxnan) [reply flink-clients/src/main/java/org/apache/flink/client/program/rest/RestClusterClient.java]: It is updated at 70ceb32 . Could you take another look?

**Pull request merged**